src/backend/cdb/cdbtm.c

```
@@ -1030,6 +1030,7 @@ doDtxPhase2Retry(void)
 				/*
 				 * KLUDGE: FtsHandleGangConnectionFailure will need a special
 				 * transaction context to tell it not to raise an ERROR...
+				 * todo: really?
```

The design here is to suppress ERROR in dispatcher, and ereport FATAL here to exit; with this patch, allocateWriterGang and createGang would possibly throw ERROR up, so the code would jump to the PG_CATCH block. Should we remove the
```
setDistributedTransactionContext(DTX_CONTEXT_QD_RETRY_PHASE_2);
```
and add a try/catch around doDispatchDtxProtocolCommand to ereport FATAL in the catch?